### ISSUE TITLE
Treatment for hidden identifications and comments

### ISSUE THREAD

**Issue by pleary** [inaturalist/INaturalistIOS#671]
The website recently released a feature where curators [can hide any identification](https://github.com/inaturalist/inaturalist/issues/3802) and [any comment](https://github.com/inaturalist/inaturalist/issues/3800) by using ModeratorActions. On the website we replace the hidden identifications and comments with a message saying "Content Hidden" which users can click to open a popover showing who hid them, when they were hidden, and the message left by the curator as to why they were hidden. Ideally mobile apps implement at least the "Content Hidden" message for hidden identifications.

I made a test observation https://www.inaturalist.org/observations/185148530 which has a hidden identification, a hidden comment, and [hidden media](https://github.com/inaturalist/INaturalistIOS/issues/670). In the [API response for this observation](https://api.inaturalist.org/v1/observations/185148530), the identification and comment have an attribute `hidden: true`. They also have an attribute `moderator_actions` which contains information on the hiding action including the hiding user, date it was hidden, and the `reason` message added by the curator when hiding the content.

Ideally hidden identifications and comments would be treated as a special case, rendering a message similar to "Content Hidden", that can be localized for the viewing user. If it makes sense to do so and there is room, the hiding user, date of hiding, and hiding `reason` could also be displayed somewhere. The hiding reason could be long, and on the web we only show it in a popover.

In summary:
- [ ] for observation identifications and comments that have an attribute `hidden: true`, do not render their content normally, rather display a message `Content Hidden` that can be localized to the viewing user's locale
- [ ] (optional) optionally include somewhere information about the user that hid the content (i.e. created the `moderator_action`), the date they were hidden, and the reason they were hidden

**Comment by alexshepard** [comment]
- [ ] show the text "Content Hidden" instead of the content
- [ ] if you tap the content, show an alert saying "Content Hidden" and " Content hidden by <moderator> on <date> because: "<reason>". there should be an OK button to dismiss the alert. if the moderated content was created by the viewer, they should also see a button to contact support, which will take them to a link that Tony will provide in slack.
